Frameworks/AudioToolbox/AudioFile.mm

```diff
@@ -580,11 +622,14 @@ DWORD AudioFileOpenWithCallbacks(void* context,
                                  AudioFile_SetSizeProc setSizeFunc,
                                  DWORD typeHint,
                                  AudioFileID _Nullable* out) {
-    EbrCallbackFile* callbackFP = new EbrCallbackFile(context, readFunc, writeFunc, getSizeFunc, setSizeFunc);
-    EbrFile* in = EbrAllocFile(callbackFP);
-
-    char header[4] = { 0 };
-    EbrFread(&header, sizeof(header), 1, in);
+    NSInputStream* in = [NSCallbackInputStream inputStreamWithCallBacks:context
+                                                               readFunc:readFunc
+                                                              writeFunc:writeFunc
+                                                            getSizeFunc:getSizeFunc
+                                                            setSizeFunc:setSizeFunc];
+    [in open];
+    char header[4] = {};
```

Comment:
>char [](start = 4, length = 4)

nit: this could be uint8_t to avoid casts.  It is unsigned char, so the other apis should work?